.circleci/config.yml

```diff
@@ -0,0 +1,269 @@
+        sudo cp -r /var/cache/autoinst ../os-autoinst
+        sudo chown -R 1000 ../os-autoinst/t/data
+      else # only in local run
+         [ -d ../os-autoinst ] || { sudo mkdir ../os-autoinst && sudo chown -R 1000 ../os-autoinst; }
+         bash .circleci/build_autoinst.sh ../os-autoinst
+       fi
+
+images:
+  - &docker_config
+    user: squamata
+    environment:
+      USER: squamata
+      PERL5LIB: lib
+      TRAVIS: yes
+      COVER_OPTS: -select_re "^/lib" -ignore_re '^t/.*' +ignore_re lib/perlcritic/Perl/Critic/Policy -coverage statement
+
+  - &base
+    image: registry.opensuse.org/devel/openqa/ci/containers/base:latest
+    <<: *docker_config
+  
+  - &dependency_bot
+    image: registry.opensuse.org/devel/openqa/ci/containers/dependency_bot:latest
+  
+jobs:
+  dependencies-pr:
+    docker:
+      - <<: *dependency_bot
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          command: |
+            git config user.email "no@email.com"
+            git config user.name "Dependency bot circleci"
+            git config core.autocrlf true
+            hub --version
+      - run:
+          command: |
+            depid=$(date +%y%m%d%H%M%S)
+            git checkout -b dependency_$depid
+            bash .circleci/build_dependencies.sh
+
+            git --no-pager diff .circleci/dependencies.txt
+            set -x
+            git diff --quiet .circleci/dependencies.txt || (
+                set +o pipefail
+                curl -s https://api.github.com/repos/os-autoinst/os-autoinst/commits | grep sha | head -n 1 | grep -o -E '[0-9a-f]{40}' > .circleci/autoinst.sha
+                git add .circleci/dependencies.txt .circleci/autoinst.sha
+                git commit -m "Dependency cron $depid"
+                git push $CIRCLE_REPOSITORY_URL dependency_$depid
+                hub pull-request -m "Dependency cron $depid" --base $CIRCLE_PROJECT_USERNAME:$CIRCLE_BRANCH
+            )
+
+  build-cache:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: ls -la && pwd 
+      - run: cat .circleci/dependencies.txt
+      - run: *chown_hack_for_cache
+      - restore_cache: *restore_cache
+      - run: *show_diagnostic_info
+      - run:
+          command: |
+            if [ ! -d /var/cache/zypp/packages/openQA ]; then
+              bash -x .circleci/build_cache.sh
+            fi
+      - save_cache: *save_cache
+
+  build-integration-cache:
+    docker:
+      - <<: *base
+    steps:
+      - checkout
+      - run: *chown_hack_for_cache_integration
+      - restore_cache: *restore_cache
+      - restore_cache: *restore_integration_cache
+      - run: find /var/cache/zypp/packages/ | wc -l
+      - run:
+          command: |
+            ls -lRa /var/cache/zypp/packages/openQA || :
+            if [ ! -d /var/cache/autoinst/t ]; then
+              bash -x .circleci/build_autoinst.sh "/var/cache/autoinst"
+            fi
+      - save_cache: *save_integration_cache
+
```

Comment:
Documentation doesn't seem to be built/published at the moment (`scripts/generate_documentation`). It's probably easy to add, but we wouldn't wanna forget about it if we decided to retire Travis after merging this PR.

Reply:
Yeah my idea was that documentation and coverarage will be added if/when we decide to switch from docker completely.

Reply:
https://progress.opensuse.org/issues/57050